addons/sale/models/sale_order.py

```diff
@@ -556,13 +556,18 @@ def _compute_expected_date(self):
                 order.expected_date = False
                 continue
             dates_list = order.order_line.filtered(
-                lambda line: not line.display_type and not line._is_delivery()
+                lambda line: not line.display_type and not line._is_delivery() and line.state != 'cancel'
             ).mapped(lambda line: line and line._expected_date())
             if dates_list:
-                order.expected_date = min(dates_list)
+                order.expected_date = order._expected_date_list_select_func()(dates_list)
```

Comment:
I don't really like handling function/methods like that.
Can't you add a method `_select_expected_date(self, expected_dates)` instead, that would be overridden in `sale_stock` to select the `min` or the `max` depending on the `picking_policy` ?